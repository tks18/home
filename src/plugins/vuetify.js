import Vue from 'vue';
import Vuetify from 'vuetify/lib';
import minifyTheme from 'minify-css-string'

Vue.use(Vuetify);

export default new Vuetify({
  theme: {
    themes: {
      light: {
        primary: '#F44336',
        secondary: '#424242',
        accent: '#82B1FF',
        error: '#FF5252',
        info: '#2196F3',
        success: '#4CAF50',
        warning: '#FFC107'
      },
    },
    options: {
      customProperties: true,
      minifyTheme: minifyTheme,
      themeCache: {
        get: key => localStorage.getItem(key),
        set: (key, value) => localStorage.setItem(key, value),
      },
    },
  },
});
