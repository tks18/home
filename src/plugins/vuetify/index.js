import Vue from 'vue';
import Vuetify from 'vuetify/lib';
import minifyTheme from 'minify-css-string';

Vue.use(Vuetify);

let themecache = JSON.parse(localStorage.getItem('themecache'));

export default new Vuetify({
  theme: {
    dark: themecache == null ? false : themecache.dark,
    themes: {
      light: {
        primary: themecache == null ? '#F44336' : themecache.theme.accent,
        secondary: '#424242',
        accent: '#82B1FF',
        error: '#FF5252',
        info: '#2196F3',
        success: '#4CAF50',
        warning: '#FFC107',
      },
      dark: {
        primary: themecache == null ? '#F44336' : themecache.theme.accent,
        secondary: '#424242',
        accent: '#82B1FF',
        error: '#FF5252',
        info: '#2196F3',
        success: '#4CAF50',
        warning: '#FFC107',
      },
    },
    options: {
      minifyTheme: minifyTheme,
      variations: false,
    },
  },
});
