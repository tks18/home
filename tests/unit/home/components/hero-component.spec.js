import Vuetify from 'vuetify';
import { vuetifyOptions } from '@p/vuetify';
import { createLocalVue, mount } from '@vue/test-utils';
import title_component from '@v/home/components/hero-component';

describe('Home Hero Component.vue', () => {
  const localVue = createLocalVue();
  let vuetify;
  let propsData;
  let wrapper;

  beforeEach(() => {
    vuetify = new Vuetify(vuetifyOptions);
    propsData = {
      ismobile: false,
    };
    wrapper = mount(title_component, {
      localVue,
      vuetify,
      propsData,
    });
  });
  it('Checks Vue instance and Matches with Snapshot', () => {
    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
