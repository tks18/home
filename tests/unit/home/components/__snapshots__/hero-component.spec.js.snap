// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Hero Component.vue Checks Vue instance and Matches with Snapshot 1`] = `
<div id="testing-id" class="column is-full non-touch" title="Testing Title">
  <div class="hero is-fullheight grad-back-light">
    <div class="hero-body ma-2 pa-8">
      <div class="container">
        <div class="columns is-multiline is-centered is-vcentered">
          <div class="column is-full">
            <div class="text text-center font-weight-bold text-h2">
              Designer, Front-end &amp; Backend Developer
            </div>
            <div class="text font-weight-bold my-2 py-2 mx-2 px-2 text-center text-h3"><span aria-haspopup="true" aria-expanded="false" class="fit-text non-touch point-cursor">

                  </span><span class="v-tooltip v-tooltip--top"><!----></span></div>
            <div class="text text-center font-weight-bold text-body-1">
              I design and code beautifully simple things, and I love what I
              do.
            </div>
          </div>
          <div class="column is-half has-text-right">
            <div class="v-avatar" style="height: 380px; min-width: 380px; width: 380px;">
              <div aria-label="Main Image" role="img" class="v-image v-responsive theme--light">
                <transition-stub name="fade-transition" mode="in-out">
                  <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
                </transition-stub>
                <div class="v-responsive__content"></div>
              </div>
            </div>
          </div>
          <div class="column is-centered is-half  text text-center text-overline font-weight-bold">
            <div class="row align-center">
              <div align="right" class="col col-6"><button type="button" class="v-btn v-btn--outlined v-btn--text theme--light elevation-24 v-size--large" raised=""><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-information-variant theme--light"></i>
                  About Me
                </span></button></div>
              <div align="left" class="col col-6"><button type="button" class="v-btn v-btn--outlined v-btn--text theme--light elevation-24 v-size--large" raised=""><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-console-network theme--light"></i>
                  What i Do
                </span></button></div>
              <div align="right" class="col col-6"><button type="button" class="v-btn v-btn--outlined v-btn--text theme--light elevation-24 v-size--large" raised=""><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-post-outline theme--light"></i>
                  My Blog
                </span></button></div>
              <div align="left" class="col col-6"><button type="button" class="v-btn v-btn--outlined v-btn--text theme--light elevation-24 v-size--large" raised=""><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-one-up theme--light"></i>
                  Some Extras
                </span></button></div>
              <div align="right" class="col col-6"><button type="button" class="v-btn v-btn--outlined v-btn--text theme--light elevation-24 v-size--large" raised=""><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-projector-screen theme--light"></i>
                  Projects
                </span></button></div>
              <div align="left" class="col col-6"><button type="button" class="v-btn v-btn--outlined v-btn--text theme--light elevation-24 v-size--large" raised=""><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-contactless-payment theme--light"></i>
                  Contact Me
                </span></button></div>
            </div>
            <!---->
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
